src/write_gcm.c

```diff
@@ -2147,57 +2147,74 @@ static monitored_resource_t *wg_monitored_resource_create(
   return NULL;
 }
 
-static monitored_resource_t *monitored_resource_create_from_fields(
-    const char *type, const char *project_id, ...) {
+
+static monitored_resource_t *monitored_resource_create_from_field_arr(
```

Comment:
I'm not too crazy about this name. How about `monitored_resource_create_from_array`?